src/inmanta/data/__init__.py

```diff
@@ -5136,15 +5136,23 @@ async def get_increment(cls, environment: uuid.UUID, version: int) -> tuple[set[
                 ores = id_to_resource[res["resource_id"]]
 
                 status = ores["status"]
-                # available -> next version
-                if status in [ResourceState.available.name]:
+                # available : next version
+                if status in [ResourceState.available.name, ResourceState.deploying.name]:
```

Comment:
This should be reverted I think.
```suggestion
                if status == ResourceState.available.name:
```